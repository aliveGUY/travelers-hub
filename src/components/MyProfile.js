import React from 'react';
import { useSelector } from 'react-redux';

export default function MyProfile() {
  const data = useSelector((state) => state.missions);
  const missions = data.filter((mission) => mission.reserved);
  return (
    <div>
      <div className="MyProfile-Missions">
        <div className="MyProfile-Missions-Wraper">
          <h1>My Missions</h1>
          <div className="MyProfile-Mission">
            {missions.map((mission) => (
              <h2 key={mission.mission_id}>{mission.mission_name}</h2>
            ))}
          </div>
        </div>
      </div>
      <div className="MyProfile-Rockets">
        <h1>My Rockets</h1>
      </div>
    </div>
  );
}
