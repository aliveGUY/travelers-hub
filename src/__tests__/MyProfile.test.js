import configureStore from 'redux-mock-store';
import {
  render, screen,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import MyProfile from '../components/MyProfile';

const mockStore = configureStore([]);

describe('Tests My profyle', () => {
  test('My profile component renders correctly/Snapshot', () => {
    const store = mockStore({
      rockets: [{
        id: '0',
        name: 'Rocket 0',
        description: 'Rocket description',
        images: ['img 0', 'img 1'],
        reserved: true,
      },
      {
        id: '1',
        name: 'Rocket 1',
        description: 'Rocket description',
        images: ['img 0', 'img 1'],
        reserved: true,
      }],
      missions: [{}],
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Reserved rockets shows', () => {
    const store = mockStore({
      rockets: [{
        id: '0',
        name: 'Rocket 0',
        description: 'Rocket description',
        images: ['img 0', 'img 1'],
        reserved: true,
      },
      {
        id: '1',
        name: 'Rocket 1',
        description: 'Rocket description',
        images: ['img 0', 'img 1'],
        reserved: true,
      }],
      missions: [{}],
    });
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );

    const element = screen.getByText('Rocket 0');

    expect(element).toBeEnabled();
  });

  test('Reserved missions shows', () => {
    const store = mockStore({
      rockets: [{}],
      missions: [{
        mission_id: '1',
        mission_name: 'Mission 1',
        description: 'Mission description',
        reserved: true,
      }],
    });
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );

    const element = screen.getByText('Mission 1');

    expect(element).toBeEnabled();
  });
});
