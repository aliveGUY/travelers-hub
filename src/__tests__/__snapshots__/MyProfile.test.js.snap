// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests My profyle My profile component renders correctly/Snapshot 1`] = `
<div
  className="d-flex"
>
  <div
    className="MyProfile-Rockets mx-5"
  >
    <h1
      className="mt-2"
    >
      My Missions
    </h1>
    <div
      className="list-group"
      onKeyDown={[Function]}
    />
  </div>
  <div
    className="MyProfile-Rockets"
  >
    <h1
      className="mt-2"
    >
      My Rockets
    </h1>
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <div
        className="h3 list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        Rocket 0
      </div>
      <div
        className="h3 list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        Rocket 1
      </div>
    </div>
  </div>
</div>
`;
